### PR TITLE
Redirect to My Collections path after deleting collection

### LINKED ISSUE #2858: When selecting "Delete Collection" from the "My Collections" page, the user should remain on the "My Collections" page.
### Descriptive summary

Currently, when on "My Collections" and "Delete Collect" is selected for any collection, after the deletion, the user is redirected to /catalog. This doesn't seem like the expected behavior from an "on page" function.

### Expected behavior

Remain on "My Collections"

### Actual behavior

Redirected to /Catalog

### Steps to reproduce the behavior

1. Select "My Collections" from the "Collections" pulldown
2. For a collection, select "Delete Collection" from the action pulldown
3. Confirm deletion

### PR DESCRIPTION
**Pull request by mjgiarlo**
Fixes #2858.

Based on #3014 and psu-stewardship/scholarsphere#401, thanks to @hortongn and @awead.

@projecthydra/sufia-code-reviewers

### CODE REVIEW

File: app/controllers/concerns/sufia/collections_controller_behavior.rb [hunk 1]
```diff
@@ -14,6 +14,20 @@ module CollectionsControllerBehavior
...
+      def after_destroy(id)
+        respond_to do |wants|
+          wants.html { redirect_to sufia.dashboard_collections_path, notice: "Collection #{id} was successfully deleted" }
+          wants.json { render_json_response(response_type: :success, message: "Collection #{id} was successfully deleted") }
+        end
+      end
+
+      def after_destroy_error(id)
+        respond_to do |wants|
+          wants.html { redirect_to sufia.dashboard_collections_path, notice: "Collection #{id} could not be deleted" }
```
Comment (jeremyf) [inline app/controllers/concerns/sufia/collections_controller_behavior.rb]: Not liking the redirect on this; It implies some semblance of success.

File: spec/controllers/collections_controller_spec.rb [hunk 2]
```diff
@@ -211,6 +211,41 @@
...
+        delete :destroy, params: { id: collection }
+        expect(response).to redirect_to(Sufia::Engine.routes.url_helpers.dashboard_collections_path)
+        expect(flash[:notice]).to eq "Collection #{collection.id} could not be deleted"
+      end
+
+      it "returns json" do
+        delete :destroy, params: { format: :json, id: collection }
+        json = JSON.parse(response.body)
+        json_description = json['description']
+        expect(json_description).to eq "Collection #{collection.id} could not be deleted"
```
Comment (jeremyf) [inline spec/controllers/collections_controller_spec.rb]: Also assert the response status

```ruby
expect(response).to have_http_status(:unprocessable_entity)
```

File: spec/controllers/collections_controller_spec.rb [hunk 3]
```diff
@@ -211,6 +211,41 @@
...
+        delete :destroy, params: { id: collection }
+        expect(response).to redirect_to(Sufia::Engine.routes.url_helpers.dashboard_collections_path)
+        expect(flash[:notice]).to eq "Collection #{collection.id} was successfully deleted"
+      end
+
+      it "returns json" do
+        delete :destroy, params: { format: :json, id: collection }
+        json = JSON.parse(response.body)
+        json_description = json['description']
+        expect(json_description).to eq "Collection #{collection.id} was successfully deleted"
```
Comment (jeremyf) [inline spec/controllers/collections_controller_spec.rb]: Please assert expected HTTP status.

**Comment by mjgiarlo** [comment]
@jeremyf The latest code incorporates changes based on your feedback. Thanks! Re-review?

**Pull request merged**